apps/web/src/components/workflow/preview/in-app/InAppPreview.tsx

```diff
@@ -39,8 +50,29 @@ export function InAppPreview({ showVariables = true }: { showVariables?: boolean
     {
       onSuccess(data) {
         setBridgeContent({
+          subject: data.outputs.subject,
           content: data.outputs.body,
-          ctaButtons: [],
+          avatar: data.outputs.avatar,
+          ctaButtons: [
+            ...(data.outputs.primaryAction
+              ? [
+                  {
+                    type: ButtonTypeEnum.PRIMARY,
+                    content: data.outputs.primaryAction.label,
+                    url: data.outputs.primaryAction.url,
+                  },
+                ]
+              : []),
+            ...(data.outputs.secondaryAction
+              ? [
+                  {
+                    type: ButtonTypeEnum.SECONDARY,
+                    content: data.outputs.secondaryAction.label,
+                    url: data.outputs.secondaryAction.url,
+                  },
+                ]
+              : []),
+          ],
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** This mutation seems worthwhile to extract to a hook with refined types and logic‏